hack/check-task-owners.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+owners_missing=0
+check_result=$(mktemp)
+
+# Check the OWNERS file is present for each task
+find task/ -mindepth 1 -maxdepth 1 -type d | \
+    while read -r task_dir; do
+        owners_file="$task_dir/OWNERS"
+        if [ ! -e "$owners_file" ]; then
+            echo "error: missing owners file $owners_file" >>"$check_result"
+            owners_missing=1
+            continue
+        fi
+        approvers=$(yq '.approvers[]' $owners_file)
+        reviewers=$(yq '.reviwers[]' $owners_file)
+        if [ -z "$approvers" ] && [ -z "$reviewers" ]; then
+            echo "$task_dir/OWNERS don't have atleast 1 approver and 1 reviewer"
+        fi
+    done
+
+if [ -s "$check_result" ]; then
+    cat "$check_result"
+    echo "Please add OWNERS file with atleast 1 approver and 1 reviewer"
+fi
+
+# Exit with non zero error code if OWNERS file is missing
+if [ "$owners_missing" -eq 1 ]; then
```

Comment:
Do we need this additional check and variable? Can't we exit in the previous "if" block? If the check_result is not empty, we are also setting the variable owners_missing=1 at the same time. So they will both be true/false at the same time, so it might be redundant if I did not miss anything.

Reply:
Good catch. Agree it can be done in a single if block as well. Fixed it now.